cloudbridge/cloud/providers/gce/services.py

```diff
@@ -267,16 +269,84 @@ def current(self):
         return self.get(self.provider.region_name)
 
 
+class GCEImageService(BaseImageService):
+
+    def __init__(self, provider):
+        super(GCEImageService, self).__init__(provider)
+
+    def get(self, image_name):
+        """
+        Returns an Image given its name
+        """
+        try:
+            image = self.provider.gce_compute \
+                                  .images() \
+                                  .get(project=self.provider.project_name,
+                                       image=image_name) \
+                                  .execute()
+            if image:
+                return GCEMachineImage(self.provider, image)
+        except TypeError as type_error:
+            # The API will throw an TypeError, if parameter `image` does not
+            # match the pattern "[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?".
+            print("TypeError: {0}".format(type_error))
+            return None
+        except googleapiclient.errors.HttpError as http_error:
+            print("googleapiclient.errors.HttpError: {0}".format(http_error))
+            return None
+
+    def find(self, name, limit=None, marker=None):
+        """
+        Searches for an image by a given list of attributes
+        """
+        filters = {'name': name}
+        try:
+            response = self.provider.gce_compute \
+                                    .images() \
+                                  .list(project=self.provider.project_name) \
+                                  .execute()
+        except googleapiclient.errors.HttpError as http_error:
+            print("googleapiclient.errors.HttpError: {0}".format(http_error))
+            return []
+        if 'items' not in response:
+            return []
+        images = [GCEMachineImage(self.provider, image) for image
+                  in response['items']
+                  if 'name' in image and image['name'] == filters['name']]
+        return ClientPagedResultList(self.provider, images,
+                                     limit=limit, marker=marker)
+
+    def list(self, limit=None, marker=None):
```

Comment:
Since the GCE API appears to supports paging directly in the API (via `maxResults` param) so we should probably be using that instead of the `ClientPagedResultList`, which is just a library-internal workaround for the APIs that don't natively support paging.